modules/Hoot/managers/layerManager.js

```diff
@@ -342,6 +343,18 @@ export default class Layers {
         this.hoot.context.flush();
     }
 
+    toggleLayerVisibility( layer ) {
+        const isVisible = layer.visible = !layer.visible,
+              id = layer.id;
+
+        if (isVisible) {
+            this.hoot.context.connection().addTile( id );
```

Comment:
I noticed that these actions cause a sort of flicker to occur when I toggle on and off. 
I think we can avoid this be just switching layer opacity between 0 and 1 for the layers' svg path elements...
Looking at the html for one of the paths...

```html
<path class="way line stroke w4_273 tag-highway tag-highway-road tag-hoot-273" d="M917.0692347958684,89.5553780272603L1010.1036382615566,90.6621542610228"></path>
```

I think that we can do a selectAll in the `.data-layer .osm` g element, then select all those with the `tag-hoot-273` classes in this example, then change opacity

Reply:
yea i tried this method as well but noticed the click handlers remained on the map. Will look into way to remove those because this approach is more efficient and none of the dumb flicker